jmx-scraper/src/main/java/io/opentelemetry/contrib/jmxscraper/JmxScraper.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.jmxscraper;
+
+import io.opentelemetry.contrib.jmxscraper.client.JmxRemoteClient;
+import io.opentelemetry.contrib.jmxscraper.config.ConfigurationException;
+import io.opentelemetry.contrib.jmxscraper.config.JmxScraperConfig;
+import io.opentelemetry.contrib.jmxscraper.config.JmxScraperConfigFactory;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.logging.Logger;
+import javax.management.MBeanServerConnection;
+import javax.management.remote.JMXConnector;
+
+public class JmxScraper {
+  private static final Logger logger = Logger.getLogger(JmxScraper.class.getName());
+
+  private final JmxRemoteClient client;
+
+  // TODO depend on instrumentation 2.9.0 snapshot
+  // private final JmxMetricInsight service;
+
+  /**
+   * Main method to create and run a {@link JmxScraper} instance.
+   *
+   * @param args - must be of the form "-config {jmx_config_path,'-'}"
+   */
+  @SuppressWarnings({"SystemOut", "SystemExitOutsideMain"})
+  public static void main(String[] args) {
+    JmxScraperConfig config;
+    JmxScraper jmxScraper = null;
+    try {
+      JmxScraperConfigFactory factory = new JmxScraperConfigFactory();
+      config = JmxScraper.createConfigFromArgs(Arrays.asList(args), factory);
+      jmxScraper = new JmxScraper(config);
+
+    } catch (ArgumentsParsingException e) {
+      System.err.println(
+          "Usage: java -jar <path_to_jmxscraper.jar> "
+              + "-config <path_to_config.properties or - for stdin>");
+      System.exit(1);
+    } catch (ConfigurationException e) {
+      System.err.println(e.getMessage());
+      System.exit(1);
+    }
+
+    try {
+      Objects.requireNonNull(jmxScraper).start();
+    } catch (IOException e) {
+      System.err.println("Unable to connect " + e.getMessage());
+      System.exit(2);
+    }
+  }
+
+  /**
+   * Create {@link JmxScraperConfig} object basing on command line options
+   *
+   * @param args application commandline arguments
+   */
+  static JmxScraperConfig createConfigFromArgs(List<String> args, JmxScraperConfigFactory factory)
+      throws ArgumentsParsingException, ConfigurationException {
+    if (!args.isEmpty() && (args.size() != 2 || !args.get(0).equalsIgnoreCase("-config"))) {
```

Comment:
I think it'd be nice to throw an exception early in the `args.isEmpty()` case. It makes this `if` simpler and prevents later trying to deal with config loading from an unspecified config. Being explicit is better here.